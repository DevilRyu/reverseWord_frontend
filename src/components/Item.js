import React from 'react'

export const Item = ({word}) => {
    return (
        <div className="border border-secondary w-75 mt-3 pl-2 pt-1 pb-1 rounded" style={{marginLeft: "10%"}}>{word}</div>
    )
}
