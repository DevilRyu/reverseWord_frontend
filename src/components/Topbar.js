import React from 'react'
import { useDispatch } from 'react-redux';
import { wordStartAddNew } from '../actions/words';

import { useForm } from '../hooks/useForm'

export const Topbar = () => {

    const dispatch = useDispatch();

    const [formValues, handleInputChange] = useForm({
        text: '',
    });

    const {text} = formValues;

    const handleSend = (e) => {
        e.preventDefault();
        dispatch(wordStartAddNew(text));
    };

    return (
        <nav className="navbar navbar-red bg-danger justify-content-center">
            <form 
                className="form-inline w-100  justify-content-center"
                onSubmit={handleSend}
            >
                <input className="form-control mr-sm-2 w-50" 
                    type="search" 
                    placeholder="Insert Text" 
                    aria-label="Search" 
                    name="text"
                    value={text}
                    onChange={handleInputChange}
                />
                <button className="btn my-2 my-sm-0 btn-outline-light bg-primary " type="submit">Send</button>
            </form>
        </nav>
    )
}
