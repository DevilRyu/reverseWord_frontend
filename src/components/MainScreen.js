import React from 'react'
import { useSelector } from 'react-redux';
import { Item } from './Item'

export const MainScreen = () => {

    const {words} = useSelector(state => state.reverseWord);
    

    return (
        <>
            <div className="card w-75" style={{ marginLeft: "13%", marginTop: "2%" }} >
                <div className="card-body">
                    <h5 className="card-title">Results: </h5>
                    <div className="card-grid">

                        {
                            words.map((word) => {
                                return <Item 
                                    key={word}
                                    word={word}/>
                            })
                        }
                    </div>
                </div>
            </div>

        </>
    )
}
