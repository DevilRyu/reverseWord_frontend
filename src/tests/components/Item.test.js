import React from 'react';
import renderer from 'react-test-renderer';
import { Item } from '../../components/Item';

describe('Pruebas en el componente - <Item />', () => {

  test('Debe de mostrarse correctamente', async () => {

    const component = renderer.create(
      <Item word="test" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });

});