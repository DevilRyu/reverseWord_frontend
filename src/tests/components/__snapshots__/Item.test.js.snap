// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente - <Item /> Debe de mostrarse correctamente 1`] = `
<div
  className="border border-secondary w-75 mt-3 pl-2 pt-1 pb-1 rounded"
  style={
    Object {
      "marginLeft": "10%",
    }
  }
>
  test
</div>
`;
